osclib/accept_command.py

```diff
@@ -32,6 +32,39 @@ def find_new_requests(self, project):
             rqs.append({'id': int(rq.get('id')), 'packages': pkgs})
         return rqs
 
+    def virtual_accept_request_has_no_binary(self, project, package):
+        pkgs = []
+        filelist = self.api.get_filelist_for_package(pkgname=package, project=self.api.project, expand='1', extension='spec')
+        for spec in filelist:
+            pkgs.append(spec[:-5])
+
+        for pkg in pkgs:
+            query = {'view': 'binarylist', 'package': pkg, 'multibuild': '1'}
+            pkg_binarylist = ET.parse(http_GET(self.api.makeurl(['build', project, '_result'], query=query))).getroot()
```

Comment:
Without running the queries to double check this seems like the same information provided by `osclib.core.package_binary_list()`. From the comment and my memory multibuild results are returned there as well. Perhaps try it and just and check if that returns an non-empty list. Otherwise the `binary_list()` method is the one used by duplicate binary check.

The main difference is the details of the information returned, but you only seem to care if any binaries or not so I would expect them to be sufficient.

Reply:
hmm... I don't want to hardcoding arch, the binary must be removed in any arch.